lib/rom/sql/attribute.rb

```diff
@@ -62,6 +62,7 @@ def canonical
       # @api public
       def qualified(table_alias = nil)
         return self if qualified? && table_alias.nil?
+        return meta(qualified: false) unless source
```

Comment:
Let's add:

``` ruby
def qualifiable?
  ! source.nil?
end
```

and do:

``` ruby
return meta(qualified: false) unless qualifiable?
```

This way it's more explicit, otherwise we're adding special meaning to `source` that is not clearly visible without reading `qualified` method and understanding what it's doing.